src/ivectorbin/compute-vad-from-frame-likes.cc

```diff
@@ -0,0 +1,215 @@
+   input of priors_str is of the form:
+      0.5,0.25,0.25
+*/
+void PreparePriors(const std::string &priors_str, int32 num_classes,
+                std::vector<BaseFloat> *priors) {
+  if (priors_str.empty()) {
+    for (int32 i = 0; i < num_classes; i++)
+      priors->push_back(log(1.0/num_classes)); // Uniform priors
+  } else {
+    SplitStringToFloats(priors_str, ",", false, priors);
+    for (int32 i = 0; i < priors->size(); i++)
+      (*priors)[i] = log((*priors)[i]);
+  }
+
+  if (priors->size() != num_classes)
+    KALDI_ERR << priors->size() << " priors specified.  Expected "
+              << num_classes;
+}
+
+}
+
+int main(int argc, char *argv[]) {
+  using namespace kaldi;
+  typedef kaldi::int32 int32;
+  try {
+    const char *usage =
+      "This program computes frame-level voice activity decisions from a\n"
+      "set of input frame-level log-likelihoods.  Usually, these\n"
+      "log-likelihoods are the output of fgmm-global-get-frame-likes.\n"
+      "Frames are assigned labels according to the class for which the\n"
+      "log-likelihood (optionally weighted by a prior) is maximal.  The\n"
+      "class labels are determined by the order of inputs on the command\n"
+      "line.  See options for more details.\n"
+      "\n"
+      "Usage: compute-vad-from-frame-likes [options] <likes-rspecifier-1>\n"
+      "    ... <likes-rspecifier-n> <vad-wspecifier>\n"
+      "e.g.: compute-vad-from-frame-likes --map=label_map.txt\n"
+      "    scp:likes1.scp scp:likes2.scp ark:vad.ark\n"
+      "See also: fgmm-global-get-frame-likes, compute-vad, merge-vads\n";
+
+    ParseOptions po(usage);
+    std::string map_rxfilename;
+    std::string priors_str;
+
+    po.Register("map", &map_rxfilename, "Table that defines the frame-level "
+      "labels.  For each row, the first field is the zero-based index of the "
+      "input likelihood archive and the second field is the associated "
+      "integer label.");
+
+    po.Register("priors", &priors_str, "Comma-separated list that specifies "
+      "the priors for each class.  The order of the floats corresponds to "
+      "the index of the input archives.  E.g., --priors=0.5,0.2,0.3");
+
+    po.Read(argc, argv);
+    if (po.NumArgs() < 3) {
+      po.PrintUsage();
+      exit(1);
+    }
+
+    unordered_map<int32, int32> map;
+    std::vector<BaseFloat> priors;
+    int32 num_classes = po.NumArgs() - 1;
+    PrepareMap(map_rxfilename, num_classes, &map);
+    PreparePriors(priors_str, num_classes, &priors);
+
+    SequentialBaseFloatVectorReader first_reader(po.GetArg(1));
+    std::vector<RandomAccessBaseFloatVectorReader *> readers;
+    std::string vad_wspecifier = po.GetArg(po.NumArgs());
+    BaseFloatVectorWriter vad_writer(vad_wspecifier);
+
+    for (int32 i = 2; i < po.NumArgs(); i++) {
+      RandomAccessBaseFloatVectorReader *reader
+        = new RandomAccessBaseFloatVectorReader(po.GetArg(i));
+      readers.push_back(reader);
+    }
+
```

Comment:
This can be done later, but I think it would be quite easy here to make it a Viterbi computation which would avoid very short runs of one or another class.  You could just specify a penalty for switching, and interpret that a log-probability for transitioning between different classes.  The computation would not be that complicated to code.  But it won't change the interfaces, so this can be done later.